File: dsp01.c
#include <graph.h>
#include <conio.h>
#include <stdio.h>
#include <math.h>

#define XDIM 320
#define YDIM 320

float   fX[XDIM + 1], fY[YDIM + 1];

void main (void)
    {
    /*
     ********  PROGRAM NO.1    DIGITAL BANDPASS FILTER  ********
     */
    int    iCount;

    _clearscreen  (_GCLEARSCREEN);
    printf ("PROGRAM 1: Digital bandpass filter\n");
    printf ("Press any key to start...\n\n");
    getch();

    /*
     *****  GENERATE INPUT SIGNAL,  AND LOAD INTO ARRAY X  *****
     */
    for (iCount = 1; iCount <= XDIM; iCount++)
        fX[iCount] = sin (3.141593 *
                         (float)iCount *
                         (0.05 + 0.0005 * (float)iCount));

    /*
     ******  ESTIMATE OUTPUT SIGNAL OF RECURSIVE FILTER  *******
     */
    fY[1] = 0.0;
    fY[2] = 0.0;

    for (iCount = 3; iCount <= YDIM; iCount++)
        fY[iCount] = 1.5 *
        fY[iCount - 1] - 0.85 *
        fY[iCount - 2] + fX[iCount];

    /*
     *******  PLOT INPUT AND OUTPUT SIGNALS ON SCREEN  *********
     */

    _setvideomode( _MRES16COLOR );


    for (iCount = 1; iCount <= XDIM; iCount++)
        {
        _moveto (iCount, 50);
        _lineto (iCount, (int)(50.0 - fX[iCount] * 25.0 + 0.5));

        _moveto (iCount, 150);
        _lineto (iCount, (int)(150.0 - fY[iCount] * 4.0 + 0.5));
        }

    getch();

    _setvideomode( _DEFAULTMODE );

    /*
     ***********************************************************
     */
    }

File: dsp01.pas
(**********  PROGRAM NO.1    DIGITAL BANDPASS FILTER  **********)
program DSP01;
uses graph3;
var X: array[1..320] of real;
var Y: array[1..320] of real;
var N: integer;
(*******  GENERATE INPUT SIGNAL, AND LOAD INTO ARRAY X  ********)
begin
  for N:=1 to 320 do
    begin
      X[N]:=sin(3.141593*N*(0.05+(0.0005*N)));
    end;
(********  ESTIMATE OUTPUT SIGNAL OF RECURSIVE FILTER  *********)
  Y[1]:=0;
  Y[2]:=0;
    for N:=3 to 320 do
      begin
        Y[N]:=1.5*Y[N-1]-0.85*Y[N-2]+X[N];
      end;
(**********  PLOT INPUT AND OUTPUT SIGNALS ON SCREEN  **********)
  graphmode;
    for N:=1 to 320 do
      begin
        draw(N,50,N,round(50-X[N]*25),1);
        draw(N,150,N,round(150-Y[N]*4),1);
      end;
end.
(***************************************************************)